test/user.test.js

```diff
@@ -1763,6 +1763,367 @@ describe('User', function() {
+                if (err) return next(err);
+                assert.equal(userInstance.firstName, 'Candy');
+                next();
+              });
+            });
+          },
+          function(next) {
+            AccessToken.find({ where: { userId: usersId }}, function(err, tokens) {
+              if (err) return next(err);
+              expect(tokens.length).to.equal(2);
+              next();
+            });
+          },
+        ], function(err) {
+          if (err) return done(err);
+          done();
+        });
+      });
+
+      it('keeps sessions AS IS if firstName is added using `updateOrCreate`', function(done) {
+        async.series([
+          function(next) {
+            User.updateOrCreate({ id: usersId, 'firstName': 'Amir',
+          'email': currentEmailCredentials.email, 'password': 'bar' },
+          function(err, userInstance) {
+            if (err) return next(err);
+            assert.equal(userInstance.firstName, 'Amir');
+            next();
+          });
+          },
+          function(next) {
+            AccessToken.find({ where: { userId: usersId }}, function(err, tokens) {
+              if (err) return next(err);
+              expect(tokens.length).to.equal(2);
+              next();
+            });
+          },
+        ], function(err) {
+          if (err) return done(err);
+          done();
+        });
+      });
+
+      it('keeps sessions AS IS if firstName is added using `replaceById`', function(done) {
+        async.series([
+          function(next) {
+            User.replaceById(usersId, { 'firstName': 'Miroslav',
+          'email': currentEmailCredentials.email, 'password': currentEmailCredentials.password },
+            function(err, userInstance) {
+              if (err) return next(err);
+              assert.equal(userInstance.email, currentEmailCredentials.email);
+              next();
+            });
+          },
+          function(next) {
+            AccessToken.find({ where: { userId: usersId }}, function(err, tokens) {
+              if (err) return next(err);
+              expect(tokens.length).to.equal(2);
+              next();
+            });
+          },
+        ], function(err) {
+          if (err) return done(err);
+          done();
+        });
+      });
+    });
+
+    describe('User not chaning email property', function() {
+      var user1Id, user2Id, user3Id;
+      it('keeps sessions untouched if email property is not changed', function(done) {
+        async.series([
+          function(next) {
+            User.create({ email: 'user1@a.com', password: 'u1pass' }, function(err, user1) {
+              User.create({ email: 'user2@b.com', password: 'u2pass' }, function(err, user2) {
+                User.create({ email: 'user3@c.com', password: 'u3pass' }, function(err, user3) {
+                  if (err) return done(err);
+                  user1Id = user1.id;
+                  user2Id = user2.id;
+                  user3Id = user3.id;
```

Comment:
Since there are only two kinds of users (a user changing the email, a user that should not be affected), it's enough to have only two user instances in the test too.